src/common/misc.c

```diff
@@ -1071,3 +1072,20 @@ daos_prop_copy(daos_prop_t *prop_req, daos_prop_t *prop_reply)
 	}
 	return rc;
 }
+
+void
+daos_dti_gen(struct dtx_id *dti, bool zero)
+{
+	static __thread struct dtx_id dtx_id_tls;
+
+	if (zero) {
+		memset(dti, 0, sizeof(*dti));
+	} else {
+		if (dtx_id_tls.dti_hlc == 0)
+			uuid_generate(dtx_id_tls.dti_uuid);
+		dtx_id_tls.dti_hlc = crt_hlc_get();
+
+		uuid_copy(dti->dti_uuid, dtx_id_tls.dti_uuid);
+		dti->dti_hlc = dtx_id_tls.dti_hlc;
+	}
+}
```

Comment:
Looks like it could be further simplified:

    static __thread uuid_t uuid;
    ...
    } else {
        if (uuid_is_null(uuid))
            uuid_generate(uuid);
        uuid_copy(dti->dti_uuid, uuid);
        dti->dti_time = crt_hlc_get();
    }

Not a problem nevertheless.

Reply:
ah, yes...   let me update it.

Reply:
Thanks.